util/dvsim/Deploy.py

```diff
@@ -203,25 +203,22 @@ def _process_exports(self):
     def _construct_cmd(self):
         """Construct the command that will eventually be launched."""
 
-        cmd = "make -f " + self.flow_makefile + " " + self.target
+        args = ["make", "-f", self.flow_makefile, self.target]
         if self.dry_run is True:
-            cmd += " -n"
+            args += ["-n"]
         for attr in sorted(self.mandatory_cmd_attrs.keys()):
             value = getattr(self, attr)
             if type(value) is list:
-                pretty_value = []
-                for item in value:
-                    pretty_value.append(item.strip())
                 # Join attributes that are list of commands with '&&' to chain
                 # them together when executed as a Make target's recipe.
                 separator = " && " if attr in self.cmds_list_vars else " "
-                value = separator.join(pretty_value)
+                value = separator.join(item.strip() for item in value)
             if type(value) is bool:
                 value = int(value)
             if type(value) is str:
                 value = value.strip()
-            cmd += " {}={}".format(attr, shlex.quote(value))
-        return cmd
+            args += ["{}={}".format(attr, shlex.quote(value))]
+        return " ".join(args)
```

Comment:
You know what I'm going to say here, but... can we just return this as a list? It would get rid of the `shlex.split` in LocalLauncher.py. You'd need to change the LsfLauncher code to something like this:
```python3
            raw_cmd = ' '.join([shlex.quote(arg) for arg in job.deploy.cmd])
            cmd = '{} >{} 2>&1'.format(raw_cmd, job.deploy.get_log_path())
```
and fix up the code in `is_equivalent_job`, but it gets rid of some string round-tripping, which is probably a good thing.